providers/flagd/pkg/service/in_process/service_file_test.go

```diff
@@ -0,0 +1,53 @@
+package process
+
+import (
+	"context"
+	of "github.com/open-feature/go-sdk/openfeature"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+)
+
+func TestInProcessOfflineMode(t *testing.T) {
+	// given
+	flagFile := "config.json"
+	offlinePath := filepath.Join(t.TempDir(), flagFile)
+
+	err := os.WriteFile(offlinePath, []byte(flagRsp), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// when
+	service := NewInProcessService(Configuration{OfflineFlagSource: offlinePath})
+
+	err = service.Init()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// then
+	channel := service.EventChannel()
+
+	select {
+	case event := <-channel:
+		if event.EventType == of.ProviderError {
```

Comment:
nitpick: to be completely sure, should we rather explicitly check for the `ProviderReady` event?